field_level_encryption/field_level_encryption_config_builder.go

```diff
@@ -0,0 +1,121 @@
+package field_level_encryption
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+)
+
+type FieldLevelEncryptionConfigBuilder struct {
+	encryptionCertificateFingerprint          string
+	oaepPaddingDigestAlgorithm                string
+	ivFieldName                               string
+	ivHeaderName                              string
+	oaepPaddingDigestAlgorithmFieldName       string
+	encryptedKeyFieldName                     string
+	encryptionCertificateFingerprintFieldName string
+	encryptionKeyFingerprintFieldName         string
+
+	encryptionCertificate    *x509.Certificate
+	encryptionKeyFingerprint string
+	decryptionKey            *rsa.PrivateKey
+	fieldValueEncoding       string
+	encryptionPaths          map[string]string
+	decryptionPaths          map[string]string
+	encryptedValueFieldName  string
+}
+
+func NewFieldLevelEncryptionConfigBuilder() *FieldLevelEncryptionConfigBuilder {
+	cb := FieldLevelEncryptionConfigBuilder{}
+	cb.encryptionPaths = make(map[string]string)
+	cb.decryptionPaths = make(map[string]string)
+
+	return &cb
+}
+
+func (cb *FieldLevelEncryptionConfigBuilder) WithEncryptionCertificate(encryptionCertificate *x509.Certificate) *FieldLevelEncryptionConfigBuilder {
+	cb.encryptionCertificate = encryptionCertificate
+	return cb
+}
+
+func (cb *FieldLevelEncryptionConfigBuilder) WithEncryptionCertificateFingerprint(encryptionCertificateFingerprint string) *FieldLevelEncryptionConfigBuilder {
+	cb.encryptionCertificateFingerprint = encryptionCertificateFingerprint
+	return cb
+}
+
+func (cb *FieldLevelEncryptionConfigBuilder) WithEncryptionKeyFingerprint(encryptionKeyFingerprint string) *FieldLevelEncryptionConfigBuilder {
+	cb.encryptionKeyFingerprint = encryptionKeyFingerprint
+	return cb
+}
+func (cb *FieldLevelEncryptionConfigBuilder) WithDecryptionKey(decryptionKey *rsa.PrivateKey) *FieldLevelEncryptionConfigBuilder {
+	cb.decryptionKey = decryptionKey
+	return cb
+}
+
+func (cb *FieldLevelEncryptionConfigBuilder) WithEncryptionPath(jsonPathIn string, jsonPathOut string) *FieldLevelEncryptionConfigBuilder {
+	cb.encryptionPaths[jsonPathIn] = jsonPathOut
+	return cb
+}
+
+func (cb *FieldLevelEncryptionConfigBuilder) WithDencryptionPath(jsonPathIn string, jsonPathOut string) *FieldLevelEncryptionConfigBuilder {
```

Comment:
Decryption is mis-spelled here

"WithDencryptionPath" -> "WithDecryptionPath"

Reply:
fixed in [3f2b1a7](https://github.com/Mastercard/client-encryption-go/pull/6/commits/3f2b1a7106e469cb1bbc24a59bd8ecb71ea6c4bc)